File: 12 - Integrando mobile e backend/GerenciadorTarefas/ios/ProjetoReactNavigation/CalculadorModule.m
//
//  CalculadorModule.m
//  ProjetoReactNavigation
//
//  Created by Douglas Nassif Roma Junior on 16/05/18.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "CalculadorModule.h"
#import <React/RCTConvert.h>

@implementation CalculadorModule

RCT_EXPORT_MODULE(Calculador);

RCT_EXPORT_METHOD(somar:(double )numA numB:(double )numB promise:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
{
  
  if (numA == 0 || numB == 0) {
    reject(@"100", @"Número não pode ser nulo.", nil);
    return;
  }
  
  double resultado = numA + numB;
  
  resolve([NSNumber numberWithDouble:resultado]);
}

@end

File: 12 - Integrando mobile e backend/GerenciadorTarefas/android/app/src/main/java/com/projetoreactnavigation/CalculadorModule.java
package com.projetoreactnavigation;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class CalculadorModule extends ReactContextBaseJavaModule {


    public CalculadorModule(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "Calculador";
    }

    @ReactMethod
    public void somar(Float numA, Float numB, Promise promise) {

        if (numA == 0 || numB == 0) {
            promise.reject("100", "Número não pode ser nulo.");
            return;
        }

        float resultado = numA + numB;

        promise.resolve(resultado);

    }

}
